src/status_im/chat/views/message/request_message.cljs

```diff
@@ -70,35 +71,35 @@
              (when command-icon
                [icon command-icon st/command-request-image])]]))})))
 
-(defn message-content-command-request
+(defview message-content-command-request
   [{:keys [message-id chat-id]}]
-  (let [commands-atom       (subscribe [:get-commands-and-responses chat-id])
-        answered?           (subscribe [:is-request-answered? message-id])
-        status-initialized? (subscribe [:get :status-module-initialized?])
-        markup              (subscribe [:get-message-preview message-id])]
-    (fn [{:keys [message-id content from incoming-group]}]
-      (let [commands @commands-atom
-            {:keys        [prefill prefill-bot-db prefillBotDb params]
+  (letsubs [requests [:chat-actions :possible-requests]
+            commands [:chat-actions :possible-commands]
+            answered? [:is-request-answered? message-id]
+            status-initialized? [:get :status-module-initialized?]
+            markup [:get-message-preview message-id]]
+    (fn [{:keys [message-id content from incoming-group] :as message}]
+      (let [{:keys        [prefill prefill-bot-db prefillBotDb params]
              text-content :text} content
-            {:keys [command content]} (parse-command-request commands content)
-            command  (if (and params command)
-                       (merge command {:prefill        prefill
-                                       :prefill-bot-db (or prefill-bot-db prefillBotDb)})
-                       command)
-            on-press-handler  (if (:execute-immediately? command)
-                                #(dispatch [:execute-command-immediately command])
-                                (when (and (not @answered?) @status-initialized?)
-                                  #(set-chat-command message-id command)))]
+            {:keys [command content]} (commands/find-command-for-request message requests commands)
+            command          (if (and params command)
+                               (merge command {:prefill        prefill
+                                               :prefill-bot-db (or prefill-bot-db prefillBotDb)})
+                               command)
+            on-press-handler (if (:execute-immediately? command)
+                               #(dispatch [:execute-command-immediately command])
+                               (when (and (not answered?) status-initialized?)
+                                 #(set-chat-command message-id command)))]
         [view st/comand-request-view
          [touchable-highlight
           {:on-press on-press-handler}
           [view st/command-request-message-view
-           (if (and @markup
```

Comment:
Huh, lots of weird derefs in existing code here. I think this might get rid of some bugs, nice one :)